contentcuration/contentcuration/viewsets/assessmentitem.py

```diff
@@ -142,3 +154,49 @@ class AssessmentItemViewSet(ValuesViewset):
     def annotate_queryset(self, queryset):
         queryset = queryset.annotate(file_ids=NotNullArrayAgg("files__id"))
         return queryset
+
+    def copy(self, pk, user=None, from_key=None, **mods):
+        try:
+            item = AssessmentItem.objects.get(assessment_id=from_key)
+        except AssessmentItem.DoesNotExist:
+            error = ValidationError("Copy assessment item source does not exist")
+            return str(error), None
+
+        if AssessmentItem.objects.filter(assessment_id=pk).exists():
+            error = ValidationError("Copy pk already exists")
+            return str(error), None
+
+        try:
+            contentnode_id = mods.pop("contentnode", None)
+
+            if not contentnode_id:
+                raise ValidationError("Field `contentnode` is required")
+
+            contentnode = ContentNode.objects.get(pk=contentnode_id)
+
+            with transaction.atomic():
+                new_item = copy.copy(item)
```

Comment:
Could you double check that files are also getting copied?

Reply:
They're aren't copied here, they're duplicated from the frontend.